### ISSUE TITLE
`UnusedPrivateMethod` not aware of using the `@PostConstruct` and `@PreDestroy` Annotations with `CDI` Managed `Bean` Classes

### ISSUE THREAD

**Issue by Pankraz76** [pmd/pmd#5737]
`UnusedPrivateMethod` not aware of using the `@PostConstruct` and `@PreDestroy` Annotations with `CDI` Managed `Bean` Classes

- https://github.com/apache/maven/pull/2334
- https://github.com/apache/maven/pull/2331
- https://github.com/apache/maven/blob/51bac9714a6deebd61589a7f8163b2e759751208/impl/maven-core/src/main/java/org/apache/maven/internal/transformation/impl/DefaultConsumerPomArtifactTransformer.java#L120
- need to check for CDI setup / teardown callbacks.
  - `@org.eclipse.sisu.PreDestroy`
  - `@org.eclipse.sisu.PostConstruct`
  - as well as java native:
    - https://docs.oracle.com/javaee/7/tutorial/cdi-basic014.htm 

<img width="1450" alt="Image" src="https://github.com/user-attachments/assets/e3dedbd8-d694-468b-8818-72e9e63455ea" />

**Comment by Pankraz76** [comment]
**Affects PMD Version:** 

**Rule:**

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_XXX_XXX.html#XXX>

**Description:**

**Code Sample demonstrating the issue:**

```java

```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by Pankraz76** [comment]
could not reproduce.

**Comment by adangel** [comment]
@Pankraz76 Next time, please use the issue template we provide for reporting false-positives.
Code samples are appreciated. Screenshots are not very useful, because we then would need to transcribe the code manually (you could have just copied the code directly).

Although your screenshot might be interesting - as this isn't a PMD issue, it is probably a [Intellij Code Inspection](https://www.jetbrains.com/help/idea/code-inspection.html)...

If you are unsure whether it's a bug, it is never wrong to first ask a question in the discussions section: https://github.com/pmd/pmd/discussions/categories/q-a

By the way: PMD's [UnusedPrivateMethod](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod) rule can be configured to recognize any CDI annotation you use. You just need to configure it.

**Comment by Pankraz76** [comment]
thanks for update. Did not know about the discussion link.
